### PR TITLE
Fix video embeds, columns, and resource link breakage

### PR DESCRIPTION
**Pull request by pushfoo**
**TL;DR:** Fixes most if not all of #2387 for `development` + add video embeds plus some copyable string literals

### Changes
There are a whole bunch.

1. Most of (if not all) of #2387 is covered, but needs a thorough check before closing the issue
2. When N < 3 for a file category, reduce the number of columns
3. Add video embedding
4. Add preformatted `":resources:sounds/jump3.wav"` literals for sounds and videos
5. Add format -> codec string tables to clean up sound and video embeds

### Follow-up work
Switch to a copy-based approach for true long-term durability.

### CODE REVIEW

**Comment by pushfoo** [comment]
Per einarf's feedback:
1. the optimal solution may be to copy our resources files into our build
2. our built-in assets total 20 megs per his estimate, but we should check
I'm currently looking at new CI failures on #2412 in doc build of all things.

**Comment by pushfoo** [comment]
**TL;DR:** Feedback from Digi is incorporated, but perfect file copying will follow later since there's loads more stuff to do

Getting folder copying "right" is less important than:
| Done? | Item | Why it's more important |
|---------|----|----------------------------------|  
| :heavy_check_mark:  |  #2403 | Doc ~~may be outright wrong~~ is now fixed |
| Pending admin account override | #2404 | RTD notifications mislead users |
|  | #2325 | Basic usability |

We also have the following upstream changes to take care of now when it's ok to break API:
| pyglet text issue | how it's broken |
|-----------------------|----------------------|
| https://github.com/pyglet/pyglet/issues/1203 | Tracking and kerning should add up for individual character adjustments |
| Font weight typing stuff in pyglet | Broken typing |
